core/src/main/java/org/axonframework/queryhandling/backpressure/DebounceBackpressure.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2010-2018. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.queryhandling.backpressure;
+
+import org.axonframework.queryhandling.UpdateHandler;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Debounce (Throttle With Timeout) backpressure mechanism - the {@code delegateUpdateHandler}  will be invoked with the
+ * last received update after which given timeout passed and there were no updates. Do note that invocation of {@code
+ * delegateUpdateHandler} will be done in separate (worker) thread.
+ * <p>
+ * Deliberate choice has to be made whether losing some updates is fine by the specific use case.
+ *
+ * @param <I> type of initial result
+ * @param <U> type of incremental updates
+ * @author Milan Savic
+ * @since 3.3
+ */
+public class DebounceBackpressure<I, U> extends TimeBasedBackpressure<I, U> {
+
+    private static final boolean INTERRUPT_RUNNING_UPDATE = false;
```

Comment:
Yes, we don't want interrupt running update. Since this constant is used to cancel the current schedule I went for affirmative naming in order not to negate it on `schedule.cancel()`. It's definitely a matter of taste, and beside before mentioned argument I don't have anything against your proposition, so I'll let you decide :)

Reply:
It's a minor thing, and taste, like you point out...let's leave it as